docs/design/openshift.md

```diff
@@ -191,4 +191,57 @@ $ cat release-manifests/image-references | head -n 10
 }
 ```
 
+## Update Image's Architecture ##
+
+The architecture of a release image is determined by the following algorithm:
+
+- check if it is a manifest list or an image by [media-types](https://github.com/openshift/docker-distribution/blob/main/docs/spec/manifest-v2-2.md#media-types). 
+
+  Here `skopeo` is chosen to show the information needed for the algorithm. It is up to the Cincinnati implementation for OpenShift how the information is retrieved. We may use `oc` (`oc image info` and `oc adm release info` to be precise) to achieve the same goal, but it would requires the understanding how `oc` gets the information. With `skopeo` (as some oc commands communicate with an OpenShift cluster), it is clear that all the information comes from the image.
+
+  ```console
+  ### manifest list
+  $ skopeo inspect --raw docker://quay.io/openshift-release-dev/ocp-release:4.15.5-multi | jq -r '.mediaType'
+  application/vnd.docker.distribution.manifest.list.v2+json
+
+  ### image manifest format
+  $ skopeo inspect --raw docker://quay.io/openshift-release-dev/ocp-release:4.15.5-s390x | jq -r '.mediaType'
+  application/vnd.docker.distribution.manifest.v2+json
+
+  $ skopeo inspect --config docker://quay.io/openshift-release-dev/ocp-release:4.18.5-s390x | jq -r '.architecture'
+  s390x
+  ```
+
+* If it is a manifest list, then the architecture is `multi`.
```

Comment:
Technically it's looking at [the number of architectures referenced by the manifest][1], so a manifest-list that only referenced a single architecture would be counted as a single-arch image.  But in practice, there's no motivation for creating a manifest-list image to only hold a single architecture, so your current phrasing will almost always be right.

[1]: https://github.com/openshift/cincinnati/blob/88bb8cb71d7309de4535864627e85dca11712ce5/cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs#L265